lib/internal/http2/compat.js

```diff
@@ -67,7 +67,7 @@ let statusConnectionHeaderWarned = false;
 // close as possible to the current require('http') API
 
 const assertValidHeader = hideStackFrames((name, value) => {
-  if (name === '' || typeof name !== 'string') {
+  if (name === '' || typeof name !== 'string' || name.indexOf(' ') > 0) {
     throw new ERR_INVALID_HTTP_TOKEN('Header name', name);
   }
```

Comment:
in http.outgoing message we do:

```js
const checkIsHttpToken = common._checkIsHttpToken;

...
if (typeof name !== 'string' || !name || !checkIsHttpToken(name)) {
  throw new ERR_INVALID_HTTP_TOKEN('Header name', name);
}
```

can we do the same thing here?

Reply:
Doing so would break (15) tests, since `checkIsHttpToken` checks for characters other than spaces as well (for eg., "content-type", which would return an `ERR_INVALID_HTTP_TOKEN`).